Tests/SwiftDiagnosticsTest/SyntaxHighlightRangeCalculatorTests.swift

```diff
@@ -0,0 +1,297 @@
+
+@_spi(Testing) import SwiftDiagnostics
+import SwiftSyntax
+import XCTest
+import _SwiftSyntaxTestSupport
+
+/// Asserts that the computed syntax highlight ranges in a given syntax tree match the expected ranges.
+///
+/// This function performs two main assertions:
+/// 1. It computes highlight ranges for a given line number within a syntax tree, using provided highlight nodes, and asserts that these match the expected ranges.
+/// 2. It asserts that the description of each provided syntax highlight node matches its expected description. This is to ensure that the correct property node has been highlighted.
+///
+/// - Parameters:
+///   - inSyntaxTree: The root syntax tree containing the code.
+///   - withHighlights: An array of tuples, each containing a syntax node and its expected description (`expectedNodeDescription`).
+///   - atLineNumber: The line number within the syntax tree for which to compute highlight ranges.
+///   - usingLineSource: An optional string representing the source code of the line being tested. If nil, the function uses the entire source string from `inSyntaxTree`.
+///   - matchExpectedRanges: An array of expected highlight ranges.
+///   - file: The file in which failure occurred.
+///   - line: The line number on which failure occurred.
+fileprivate func assertSyntaxHighlightRanges(
+  inSyntaxTree tree: some SyntaxProtocol,
+  withHighlights highlights: [(node: any SyntaxProtocol, expectedNodeDescription: String)],
+  atLineNumber lineNumber: Int = 1,
+  usingLineSource lineSource: String? = nil,
+  matchExpectedRanges expectedRanges: [Range<Int>],
+  file: StaticString = #file,
+  line: UInt = #line
+) {
+  let sourceLocationConverter = SourceLocationConverter(
+    fileName: "<SyntaxHighlightRangeCalculatorTests>",
+    tree: tree
+  )
+
+  let computedRanges = SyntaxHighlightRangeCalculator.computeHighlightRanges(
+    forLine: (highlights: highlights.map { Syntax(fromProtocol: $0.node) }, sourceString: lineSource ?? tree.description),
+    at: lineNumber,
+    fromTree: tree,
+    usingSourceLocationConverter: sourceLocationConverter
+  )
+
+  XCTAssertEqual(computedRanges, expectedRanges, file: file, line: line)
+  for highlight in highlights {
+    assertStringsEqualWithDiff(
+      highlight.node.description,
+      highlight.expectedNodeDescription,
+      """
+      The description of the passed highlight node of type \(highlight.node.syntaxNodeType) does not \
+      align with the expected description.
+      """,
+      additionalInfo: """
+        Expected:
+        \(highlight.expectedNodeDescription)
+
+        Actual:
+        \(highlight.node.description)
+        """,
+      file: file,
+      line: line
+    )
+  }
+}
+
+final class SyntaxHighlightRangeCalculatorTests: XCTestCase {
+  func testComputeHighlightRangesFirstHighlightExcludesLeadingTrivia() {
+    let decl: DeclSyntax =
+      "@PropertyWrapper private(set) var variable: Decimal = 123 * 456"
+    // 123456789012345678901234567890
```

Comment:
Instead of adding this comment, what do you think about changing `assertSyntaxHighlightRanges` to actually underline the highlighted ranges? Either using a `DiagnosticDecorator` or using a hand-rolled function? I just think that it would make reading the tests a bit easier.